### PR TITLE
Skip Parquet encryption read tests if Parquet version is less than 1.12 [databricks]

### LINKED ISSUE #5959: [BUG] test_parquet_read_encryption fails
**Describe the bug**
When I run the python integration tests on Spark 3.2.0 or spark_3.3.0

```
env -u SPARK_CONF_DIR SPARK_HOME=~/spark_3.2.0/ ./run_pyspark_from_build.sh -k 'parquet and encryption'
```

Most of them fail with an error like 

```
Caused by: java.lang.ClassNotFoundException: Class org.apache.parquet.crypto.keytools.mocks.InMemoryKMS not found
	at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2571)
	at org.apache.parquet.hadoop.util.ConfigurationUtil.getClassFromConfig(ConfigurationUtil.java:33)
```

All I did was to run buildall.sh and then cd into the integration tests directory and run the tests.

I looks like 

```
@pytest.mark.skipif(os.environ.get('INCLUDE_PARQUET_HADOOP_TEST_JAR', 'false') == 'false', reason='INCLUDE_PARQUET_HADOOP_TEST_JAR is disabled')
```

is the culprit.  It is being set to true in the run_pyspark_from_build.sh without ever even checking if the hadoop dependency was placed where it is expected to be. But honestly I have no idea because I don't know what jar is expected to be on the classpath. It could be the fact that I build all of the versions and somehow the wrong dependency was downloaded. Either way this is a bad user experience.

### PR DESCRIPTION
**Pull request by razajafri**
This PR checks the Parquet version by parsing the name of the jar located in `project_root/integration_tests/target/dependency/` 

fixes #5959 

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

### CODE REVIEW

**Comment by razajafri** [comment]
build

File: integration_tests/run_pyspark_from_build.sh [hunk 1]
```diff
@@ -57,7 +57,10 @@ else
     else
         AVRO_JARS=$(echo "$SCRIPTPATH"/target/dependency/spark-avro*.jar)
         PARQUET_HADOOP_TESTS=$(echo "$SCRIPTPATH"/target/dependency/parquet-hadoop*.jar)
-        export INCLUDE_PARQUET_HADOOP_TEST_JAR=true
+        REGEX="parquet-hadoop-1\.([0-9]{2})"
+        [[ $PARQUET_HADOOP_TESTS =~ $REGEX  ]]
+        # Make sure we have Parquet version >= 1.12 in the dependency
+        export INCLUDE_PARQUET_HADOOP_TEST_JAR=$([ "${BASH_REMATCH[1]}" -ge "12" ] && echo true || echo false)
```
Comment (jlowe) [inline integration_tests/run_pyspark_from_build.sh:63]: This is going to fail when the Parquet version goes to 2.x or later.  We could leverage `sort -V` to solve the sorting issue, doing something like this:
```suggestion
        MIN_PARQUET_JAR="$SCRIPTPATH/target/dependency/parquet-hadoop-1.12.0-tests.jar"
        # Make sure we have Parquet version >= 1.12 in the dependency
        LOWEST_PARQUET_JAR=$(echo -e "$MIN_PARQUET_JAR\n$PARQUET_HADOOP_TESTS | sort -V | head -1)
        export INCLUDE_PARQUET_HADOOP_TEST_JAR=$([[ "$LOWEST_PARQUET_JAR" == "MIN_PARQUET_JAR" ]] && echo true || echo false)
```

Comment (razajafri) [reply integration_tests/run_pyspark_from_build.sh:63]: Thank you. This is much better and future proof

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]
Looks OK to me, but it would be good to get another opinion.

**Comment by razajafri** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**